src/cursor/cur_std.c

```diff
@@ -757,6 +864,14 @@ __wt_cursor_init(WT_CURSOR *cursor,
 		F_SET(cursor, WT_CURSTD_RAW);
 
 	/*
+	 * WT_CURSOR.modify supported on 'u' value formats, but may have been
+	 * already initialized.
+	 */
+	if (WT_STREQ(cursor->value_format, "u") &&
```

Comment:
I think this raises an interesting question about whether we need another dimension in our API, allowing applications to apply update-vectors for multiple elements in a more complex schema. @michaelcahill thoughts?